### PR TITLE
Update SystemVM scripts from python2 to python3

### PR DESCRIPTION
**Pull request by GabrielBrascher**
## Description

- Considering that Python 2.7 will not be maintained in 2020 (will be retired on 1st January 2020) next VR template should have Python3 installed to support python3 scripts; additionally, VR's Python2 codebase should be migrated to Python3.

- This PR covers partially issue #3195.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by DaanHoogland** [comment]
some pylint errors in travis job 1, mostly requiring extra blank lines at places @GabrielBrascher .

File: engine/components-api/src/main/java/com/cloud/ha/HighAvailabilityManager.java [hunk 1]
```diff
@@ -28,6 +30,9 @@
  * HighAvailabilityManager checks to make sure the VMs are running fine.
  */
 public interface HighAvailabilityManager extends Manager {
+
```
Comment (DaanHoogland) [inline engine/components-api/src/main/java/com/cloud/ha/HighAvailabilityManager.java]: @GabrielBrascher why do I see java changes in this PR?

Comment (GabrielBrascher) [reply engine/components-api/src/main/java/com/cloud/ha/HighAvailabilityManager.java]: My bad @DaanHoogland, I was debugging this and did commit by mistake, will soon be removed soon.

**Comment by DaanHoogland** [comment]
@GabrielBrascher can you give an indication on this. as we are out of time for python2 now?
how much more work needed and should it be reviewed yet!

**Comment by GabrielBrascher** [comment]
@DaanHoogland it still needs some work and testing, therefore, we can wait a bit for reviewing it.
I will be testing it tomorrow to re-assess the state; the plan is to address the pending points until the end of the week.

**Comment by DaanHoogland** [comment]
ping @GabrielBrascher , just remove the WIP tag and ping me, when you are done.

**Comment by GabrielBrascher** [comment]
Removing WIP tag, @DaanHoogland. I have been doing some tests and things are looking good now for the review. Still, I am testing and looking for corner case issues, but the overall workflow is working and VMs are deployed on a network with python3 scripts on VR.

**Comment by DaanHoogland** [comment]
@wido @weizhouapache @PaulAngus @rhtyd @andrijapanicsb @swill @pdion891 @kiwiflyer 
This looks like a big risk to put in now, but after 4.14 in conjunction with the template on debian10 effort?

**Comment by weizhouapache** [comment]
> @wido @weizhouapache @PaulAngus @rhtyd @andrijapanicsb @swill @pdion891 @kiwiflyer
> This looks like a big risk to put in now, but after 4.14 in conjunction with the template on debian10 effort?

agree.

**Comment by rhtyd** [comment]
@GabrielBrascher since we're closer to the release already I've moved this to 4.15 where we can do both migration to py3 as well as new Debian 10 based systemvmtemplate (#3797)

**Comment by wido** [comment]
@rhtyd I understand the postpone. We should however make sure this makes it into 4.15 as Python 2 is now EOL.

**Comment by GabrielBrascher** [comment]
Got it @rhtyd @DaanHoogland. I understand that this PR has its risks and it requires testing effort. Let's merge it on the next master branch then.

**Comment by rhtyd** [comment]
Yes absolutely @wido. Thanks @GabrielBrascher let's do this and migrate with a new Debian 10 based systemvmtempalte with Python3 the first thing after 4.14 branch/RC is cut and master is opened for changes.

**Comment by DaanHoogland** [comment]
this should be coordinated with #3797

**Comment by rhtyd** [comment]
@GabrielBrascher any update for us, you still working on this? Thanks.

**Comment by GabrielBrascher** [comment]
Sorry for the late reply, @rhtyd. I just fixed conflicts and ported to the current master.
I am now performing tests and if all goes right I shall have this PR ready for review soon.

**Comment by GabrielBrascher** [comment]
**UPDATE:** After updating the code I was able to build the System VM template and test it (on KVM).

**Environment:** KVM
**Network type:** Advanced with security groups

**Test -** Deploy VMs and assert that User VM gets IPv6 address correctly:
1. Deploy VM on a network (let's call this network of _python3_) with no VR
2. VR is deployed, health check scripts are all good
3. User VM is deployed on the _Python3_ network
4. SSH on IPv6 address works

Based on the test result, this PR I am moving it from "_Draft_" to "_ready for review_". With that said, I encourage reviewers to check the code and test SystemVMs with the Python3 changes ;-)

**Note:** there are still pending tests such as VR for Basic networks, Advanced networks without SG, VPCs, etc. Additionally, I have not tested this template on SSVM or CPVM.

**Comment by GabrielBrascher** [comment]
Travis is failing due to `pycodestyle` failures: https://travis-ci.org/github/apache/cloudstack/jobs/715011759
I will address these "checkstyle" issues soon and update the PR.

**Comment by GabrielBrascher** [comment]
Pycodestyle has passed after last commit. Jenkins and Travisa are looking good.
This one is ready for review/tests.

**Comment by GabrielBrascher** [comment]
Current Travis Pylint errors are related to the fact of Pylint evaluating a, supposedly, Python2 code.

```
pylint  --disable=R,C,W configure.py 
No config file found, using default configuration
************* Module configure
E: 24, 0: No name 'request' in module 'urllib' (no-name-in-module)
E: 24, 0: Unable to import 'urllib.request' (import-error)
E: 25, 0: No name 'parse' in module 'urllib' (no-name-in-module)
E: 25, 0: Unable to import 'urllib.parse' (import-error)
E: 26, 0: No name 'error' in module 'urllib' (no-name-in-module)
E: 26, 0: Unable to import 'urllib.error' (import-error)
E: 81,23: Undefined variable 'urllib' (undefined-variable)
E: 82,26: Undefined variable 'urllib' (undefined-variable)
E: 84,27: Undefined variable 'urlopen' (undefined-variable)

-------------------------------------------------------------------
Your code has been rated at 9.39/10 (previous run: 10.00/10, -0.61)

```

If Travis had pylint3, then it would be evaluating a Python3 code. I get the following result on pylint3

```
pylint3  --disable=R,C,W configure.py 
No config file found, using default configuration

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 9.39/10, +0.61)

```

:arrow_down: The same applies for the errors raised by Travis regarding `passwd_server_ip.py`:
```
pylint  --disable=R,C,W passwd_server_ip.py 
No config file found, using default configuration
************* Module passwd_server_ip
E: 34, 0: No name 'parse' in module 'urllib' (no-name-in-module)
E: 34, 0: Unable to import 'urllib.parse' (import-error)
E: 36, 0: Unable to import 'http.server' (import-error)
E: 37, 0: Unable to import 'socketserver' (import-error)

-------------------------------------------------------------------
Your code has been rated at 8.60/10 (previous run: 10.00/10, -1.40)
```
vs pylint3
```
pylint3  --disable=R,C,W passwd_server_ip.py 
No config file found, using default configuration

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 8.60/10, +1.40)
```

**Comment by GabrielBrascher** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@GabrielBrascher can you hacve a look at the conflict?

**Comment by GabrielBrascher** [comment]
@DaanHoogland updated this PR fixing the conflict with master.

**Comment by GabrielBrascher** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@GabrielBrascher I see pylint errors again in travis job 1, can you have a look?

**Comment by DaanHoogland** [comment]
@GabrielBrascher @wido @rhtyd @weizhouapache @PaulAngus python3 work in general has been neglected for a while. Are we still going to push it for 4.15?
@blueorangutan package

**Comment by GabrielBrascher** [comment]
@DaanHoogland I have been able to run some tests on the Advanced network with security groups, and a few tests on the basic network. But it still lacking some tests on advanced networks with VPC, etc.

With that said, I think that we will need a couple more months and also more review/test.

**Comment by DaanHoogland** [comment]
@GabrielBrascher that would mean moving to the next milestone!

**Comment by GabrielBrascher** [comment]
@DaanHoogland correct! I migrated it to 4.16 so we don't wait longer to release 4.15.

**Comment by GabrielBrascher** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@GabrielBrascher can we close this?

**Comment by GabrielBrascher** [comment]
@DaanHoogland sure! I will close this one, if necessary we can always re-open it.

**Comment by GabrielBrascher** [comment]
PR closed in favor of #4727.
